modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/taglib/servlet/taglib/CommerceOrderNotesScreenNavigationEntry.java

```diff
@@ -61,6 +68,31 @@ public String getScreenNavigationKey() {
 			SCREEN_NAVIGATION_KEY_COMMERCE_ORDER_GENERAL;
 	}
 
+	@Override
+	public boolean isVisible(User user, CommerceOrder commerceOrder) {
+		boolean hasPermission = false;
+		PermissionChecker permissionChecker =
+			PermissionCheckerFactoryUtil.create(user);
```

Comment:
Why not simply use the usual `PermissionThreadLocal.getPermisionChecker()` method, I see it's mostly used in tests?

If you want to make good use of the `User` there's also the
```
public static PermissionChecker getPermissionChecker(User user, boolean setThreadLocal)
```

We have examples of it in our `CommerceOrderModelResourcePermissionLogic` for instance.